### PR TITLE
Adding buddy list Filtering

### PR DESCRIPTION
**Pull request by danielriddell21**
#452 
Buddy List filtering as it is very noisy and Spammy at the moment

### CODE REVIEW

**Comment by bobland1** [comment]
I have started work on this. Very minor as of now but I'll look more into this. Hopefully I'll manage to work something out.

**Comment by Schmavery** [comment]
@bobland1 Looks like you're making these changes though the web interface, so I just wanted to check if you know that it's possible to sync the repo to your local machine.
When you do this, then you can hook up whatever code you have that's using the api to use your local version instead fairly easily, and in that way, test if it's working.
Feel free to ignore if you already know all this 😛

File: src/listen.js [hunk 1]
```diff
@@ -121,21 +121,27 @@ module.exports = function(defaultFuncs, api, ctx) {
...
-                for(var userID in v.buddyList) {
-                  var formattedPresence = utils.formatProxyPresence(v.buddyList[userID], userID);
-                  if(formattedPresence != null)
-                  {
-                    globalCallback(null, formattedPresence);
-                  }
-                }
-                return;
-              }
+              case 'buddylist':
```
Comment (Schmavery) [inline src/listen.js]: Not sure what's happening here, I think maybe you misunderstood me before.
You should be able to simply replace
```
if(!ctx.globalOptions.updatePresence) {
```

with something like 

```
if(!ctx.globalOptions.updatePresence || ctx.globalOptions.updateBuddyList) {
```

**Comment by bobland1** [comment]
Yeah, I'm updating through the web interface due to being on my phone then. In the morning when I'm on my PC I can do all the pushing and syncing

**Comment by bobland1** [comment]
And yeah I seem to have misunderstood 😂

**Comment by Schmavery** [comment]
I've been trying to think about how to offer the best api here and I think we should tweak this design a little.

Let's call it `batchPresence` and have it turned off by default.  Then we can have it so that if you have `updatePresence` *and* `batchPresence` turned on, you will get the new events.

The reason we will call it `batchPresence` is that now that this is behind a flag (so it's not a breaking change) we can update the signature of the presence object to return an array if there are multiple elements.

@GAMELASTER sorry for the back and forth on the signature here.  Any chance you can help @bobland1 with this change?

**Comment by bobland1** [comment]
image

File: src/listen.js [hunk 2]
```diff
@@ -135,11 +135,10 @@ module.exports = function(defaultFuncs, api, ctx) {
                 }
                 return;
               }
-
               break;
             case 'buddylist_overlay':
               // TODO: what happens when you're logged in as a page?
-              if(!ctx.globalOptions.updatePresence) {
+              if(!ctx.globalOptions.batchPresence) {
```
Comment (GAMELASTER) [inline src/listen.js]: Make here to if the updatePresence and batchPresence must be a true

Comment (bobland1) [inline src/listen.js]: Okay? I'll change that. Little confused as to why but will change now

Comment (bobland1) [inline src/listen.js]: Added minor adjustments and fixed as you correctly pointed out updatePresence. Told ya I was prone to typos 😄

**Comment by GAMELASTER** [comment]
Well, this looks well and okay! :blush:

**Comment by bobland1** [comment]
Thank you :)

File: src/listen.js (hunk 2 above)
Comment (Qualitymix) [reply src/listen.js]: I'm having trouble keeping a persistent friend list that is accurate.  Every 2 minutes or so, Facebook will tell me a whole bunch of people who are actually online, are now offline.  I'm assuming in this case the 0 means their status hasn't changed.     This is code fix that?

Comment (Qualitymix) [reply src/listen.js]: so how can i properly use batchPresence?

**Pull request closed without merging**